src/store/model.ts

```diff
@@ -187,6 +188,68 @@ export interface IActiveTool<T extends TToolType = TToolType> {
   state: TToolState<T>;
 }
 
+export enum ProgressType {
+  ANNOTATION_FETCH = "ANNOTATION_FETCH",
+  ANNOTATION_SAVE = "ANNOTATION_SAVE",
+  ANNOTATION_DELETE = "ANNOTATION_DELETE",
+  ANNOTATION_COMPUTE = "ANNOTATION_COMPUTE",
+  PROPERTY_FETCH = "PROPERTY_FETCH",
+  PROPERTY_COMPUTE = "PROPERTY_COMPUTE",
+  CONNECTION_FETCH = "CONNECTION_FETCH",
+  CONNECTION_SAVE = "CONNECTION_SAVE",
+  CONNECTION_DELETE = "CONNECTION_DELETE",
+  VIEW_FETCH = "VIEW_FETCH",
+  LAYER_CACHE = "LAYER_CACHE",
+  QUADTILE_CACHE = "QUADTILE_CACHE",
+  MAXMERGE_SCHEDULE = "MAXMERGE_SCHEDULE",
+  MAXMERGE_CACHE = "MAXMERGE_CACHE",
+  HISTOGRAM_SCHEDULE = "HISTOGRAM_SCHEDULE",
+  HISTOGRAM_CACHE = "HISTOGRAM_CACHE",
+  GENERIC = "GENERIC",
+}
+
+export const PROGRESS_TYPE_ORDER = new Map<ProgressType, number>([
```

Comment:
After asking my friend Claude :), I think the Map is a bit more flexible long term in case I want to separate types from ordering for some reason. Like, if I wanted to have two types have the same order for some reason.